bcipy/signal_model/mach_learning/classifier/function_classifier.py

```diff
@@ -185,3 +186,114 @@ def fit_transform(self, x, y, p=[]):
 
         return self.transform(x)
 
+
+class MDiscriminantAnalysis:
+    """ Robust version of RDA where every estimated statistic is estimated
+    Ref:
+        Friedman, Jerome H. "Regularized discriminant analysis."
+        Journal of the American statistical association 84.405 (1989): 165-175
+
+        and
+
+        Kadioglu, Berkan "M Estimation Based Robust Subspace learning for Brain Computer Interfaces"
+
+    Attr:
+        lam(float): shrinkage param
+        gam(float): threshold param (a.k.a. regularization param)
+        means(list): Means' of each channel
+        covariances(list): Covariances' of each channel
+        S_matrices(list): Covariance times N for each class
+        S(list): Covariance times N for all data
+        N_list(list): List of number of samples of each class
+        class_list(list): List of classes
+        priors(list): List of priors for each class
+        inv_reg_covariances(list): List of inverse regularized covariances for each class
+
+    """
+    def __init__(self):
+        # means and covariances of each channel with the order inherent in data.
+        self.means = []
+        self.covariances = []
+        self.S_matrices = []  # Covariance times N for each class
+        self.S = []
+        self.N_list = []  # List of number of samples of each class
+        self.class_list = []
+        self.priors = []
+
+        self.inv_reg_covariances = []  # Inverse regularized covariances
+
+        for z in range(10):  # add an empty list to the list will store every folds stats.
+            self.means.append([])
+            self.covariances.append([])
+            self.S_matrices.append([])
+            self.S.append([])
+            self.N_list.append([])
+            self.priors.append([])
+
+            self.inv_reg_covariances.append([])
+
+        self.lam = .9
+        self.gam = .1
+
+        self.current_fold = -1
+
+    def fit(self, x, y):
+        """
+        :list x: data, each element is every channel's trials.
+        """
+        N, p = x.shape  # N is number of samples, p is number of features
```

Comment:
lower case and no single variable names

Reply:
See my comment above for namings. Capital n is generally number of samles and p is number of predictors, i.e. features, i.e. dimensions.

Reply:
Mathematical notation != appropriate variable naming

Reply:
Changed N and p.